framework/tlvf/src/include/tlvf/CmduMessage.h

```diff
@@ -20,133 +21,40 @@ namespace ieee1905_1 {
 class CmduMessage {
 
 public:
-    CmduMessage();
-    ~CmduMessage();
+    CmduMessage() = delete;
+    CmduMessage(uint8_t *buff, size_t buff_len) : msg(buff, buff_len){};
+    ~CmduMessage(){};
 
-public:
-    std::shared_ptr<cCmduHeader> getCmduHeader() const;
-
-    template <class T, class = typename std::enable_if<std::is_base_of<BaseClass, T>::value>::type>
-    std::shared_ptr<T> addClass()
-    {
-
-        auto last_tlv = m_class_vector.empty() ? nullptr : m_class_vector.back();
-        // do not allow to use addClass method if the last tlv is not fully initialized
-        if (last_tlv) {
-            if (!last_tlv->isPostInitSucceeded()) {
-                TLVF_LOG(ERROR) << "TLV post init failed";
-                return nullptr;
-            }
-        }
-
-        std::shared_ptr<T> ptr;
-        if (m_cmdu_header) {
-            if (m_class_vector.size() == 0) {
-                ptr = std::make_shared<T>(m_cmdu_header, m_parse, m_swap);
-            } else {
-                ptr = std::make_shared<T>(m_class_vector.back(), m_parse, m_swap);
-            }
-            if (!ptr || ptr->isInitialized() == false) {
-                return std::shared_ptr<T>();
-            } else {
-                m_class_vector.push_back(std::shared_ptr<BaseClass>(ptr));
-            }
-        } else {
-            return std::shared_ptr<T>();
-        }
-        return ptr;
-    }
+    ClassList msg;
```

Comment:
>I don't understand why did you make ClassList be a member of CmduMessage instead of just be it.

Don't understand what you mean by "just be it." - the __whole__ point in this change is to decouple the TLVs from CmduMessage class, which is why we crearted `ClassList` and put it as a member of `CmduMessage`.
It also makes perfect sense - a Cmdu Message is a descriptor of TLVF generated classes, so it includes a member ClassList which holds all the TLVs, where the first class is the CmduHeader and the rest are the TLVs themselves.

>Also, why did you choose to use a member over heritage? It would have save you all the wrapper functions.

The wrapper functions are only for backward compatibility. We chose member over inheritance cause it makes more sense and we don't want CmduMessage to access all the the private and protected members of ClassList.